webui/react/src/utils/browser.ts

```diff
@@ -16,3 +19,63 @@ export const getCookie = (name: string): string | null => {
   const value = document.cookie.replace(regex, '$1');
   return value ? value : null;
 };
+
+const downloadBlob = (filename: string, data: Blob): void => {
+  const url = window.URL.createObjectURL(data);
+  const element = document.createElement('a');
+  element.setAttribute('download', filename);
+  element.style.display = 'none';
+  element.href = url;
+  document.body.appendChild(element);
+  element.click();
+  window.URL.revokeObjectURL(url);
+  document.body.removeChild(element);
+};
+
+// Different JS engines impose different maximum string lenghts thus for downloading
+// large text we it into different parts and then stich them together as a Blob
+export const downloadText = (filename: string, parts: BlobPart[]): void => {
+  const data = new Blob(parts, { type: 'text/plain' });
+  downloadBlob(filename, data);
+};
+
+export const downloadTrialLogs = (trialId: number): Promise<void> => {
+  // string concatnation is fater than array.join but is also more memory friendly?
+  // https://jsben.ch/OJ3vo
+  // characters are utf16 encoded. v8 has different internal representations.
+  const MAX_PART_SIZE = 128 * Math.pow(2, 20); // 128m * CHAR_SIZE
+  const parts: BlobPart[] = [];
+  let downloadStringBuffer = '';
+  return consumeStream<DetSwagger.V1TrialLogsResponse>(
+    DetSwagger.ExperimentsApiFetchParamCreator().determinedTrialLogs(trialId),
+    (ev) => {
+      if (downloadStringBuffer.length > MAX_PART_SIZE) {
+        parts.push(downloadStringBuffer);
+        downloadStringBuffer = '';
+      }
+      downloadStringBuffer += ev.message;
+    },
+  ).then(() => downloadText(`trial_${trialId}_logs.txt`, parts));
```

Comment:
we need experiment details here to include in the filename to keep parity with the old download